### PR TITLE
Upgrade to arc49

### PR DESCRIPTION
**Pull request by dkent600**
Upgrade to arc49

See [@daostack/Arc@0.0.0-alpha.49 release](https://github.com/daostack/arc/releases/tag/0.0.0-alpha.49).

- new function `Redeemer.redeemables()` that returns information about what would happen if you called `Redeemer.redeem()`.
- `GenesisProtocolWrapper.getProposalStatus` return values have changed
- cleaned up use of `TxGeneratingFunctionOptions` interface

**Breaking Changes**
- `GenesisProtocolWrapper.getVoteStake` is gone, replaced by new fields in `GenesisProtocolWrapper.getProposalStatus`
- `GenesisProtocolWrapper.getProposalStatus` return values have changed
- `totalVotes` has been removed from `GenesisProtocolWrapper.GenesisProtocolProposal`
- The following `GenesisProtocolWrapper` functions have been removed:
  - getRedeemableReputationProposer
  - getRedeemableTokensVoter
  - getRedeemableReputationVoter
  - getRedeemableTokensStaker
  - getRedeemableTokensStakerBounty
  - getRedeemableReputationStaker

### CODE REVIEW

File: test/redeemer.ts [hunk 1]
```diff
@@ -62,22 +70,116 @@ describe("Redeemer", () => {
...
+    const gp = (await dao.getSchemes("GenesisProtocol"))[0].wrapper as GenesisProtocolWrapper;
+    await helpers.vote(result.votingMachine, proposalId, BinaryVoteResult.No, accounts[2]);
+    await helpers.vote(result.votingMachine, proposalId, BinaryVoteResult.Yes, accounts[3]);
+
+    await (await gp.stakeWithApproval({ amount: web3.toWei(10), vote: 1, proposalId })).getTxMined();
+
     await helpers.vote(result.votingMachine, proposalId, BinaryVoteResult.Yes, accounts[1]);
 
-    await helpers.increaseTime(1);
+    await helpers.increaseTime(259200);
```
Comment (orenyodfat) [inline test/redeemer.ts]: add a comment here that this is the boosting timeout default value.

File: lib/wrappers/genesisProtocol.ts [hunk 2]
```diff
@@ -1241,67 +1049,19 @@ export interface GenesisProtocolParams extends TxGeneratingFunctionOptions {
 }
 
 export interface GetGenesisProtocolParamsResult {
-  /**
-   * The time limit in seconds for a proposal to be in relative voting mode.
```
Comment (tibetsprague) [inline lib/wrappers/genesisProtocol.ts]: why removing comments here?

**Comment by orenyodfat** [comment]
Why schemes wrappers which are not related to this pr have been changed ?

File: lib/wrappers/redeemer.ts [hunk 3]
```diff
@@ -41,6 +42,54 @@ export class RedeemerWrapper extends ContractWrapperBase {
...
+        throw new Error(ex.message);
+      });
+
+    return {
+      contributionRewardEther: result[3][2],
+      contributionRewardExternalToken: result[3][3],
+      contributionRewardNativeToken: result[3][1],
+      contributionRewardReputation: result[3][0],
+      daoBountyPotentialReward: result[1][1],
+      daoBountyReward: result[1][0],
```
Comment (tibetsprague) [inline lib/wrappers/redeemer.ts]: this is the stake bounty?

File: lib/wrappers/redeemer.ts [hunk 4]
```diff
@@ -41,6 +42,54 @@ export class RedeemerWrapper extends ContractWrapperBase {
...
+      .catch((ex: Error) => {
+        throw new Error(ex.message);
+      });
+
+    return {
+      contributionRewardEther: result[3][2],
+      contributionRewardExternalToken: result[3][3],
+      contributionRewardNativeToken: result[3][1],
+      contributionRewardReputation: result[3][0],
+      daoBountyPotentialReward: result[1][1],
```
Comment (tibetsprague) [inline lib/wrappers/redeemer.ts]: this is max stakes bounty?

File: lib/wrappers/tokenCapGC.ts [hunk 5]
```diff
@@ -58,7 +59,7 @@ export class TokenCapGCWrapper extends ContractWrapperBase {
 export const TokenCapGCFactory =
   new ContractWrapperFactory("TokenCapGC", TokenCapGCWrapper, new Web3EventService());
 
-export interface TokenCapGcParams extends TxGeneratingFunctionOptions {
```
Comment (tibetsprague) [inline lib/wrappers/tokenCapGC.ts]: why did you switch from extends to merging the interfaces?

**Comment by dkent600** [comment]
@orenyodfat 

> Why schemes wrappers which are not related to this pr have been changed ?

This was a change that I wanted to make in the code that is directly relevant to this PR, and it made sense to make the equivalent changes everywhere else where needed the code.  The changes are only with typings where the only risk is at code compile type.

File: test/redeemer.ts (hunk 1 above)
Comment (dkent600) [reply test/redeemer.ts]: done

**Pull request merged**